giskard_vision/image_classification/dataloaders/loaders.py

```diff
@@ -76,3 +77,85 @@ def get_labels(self, idx: int) -> Optional[np.ndarray]:
         row = self.get_row(idx)
 
         return np.array([row[self.label_key]])
+
+
+class DataLoaderSkinCancerHuggingFaceDataset(DataLoaderHuggingFaceDataset):
+    """
+    A data loader for the `marmal88/skin_cancer` dataset on HF, extending the DataLoaderHuggingFaceDatasets class.
+
+    Attributes:
+        label_key (str): Key for labels in the dataset.
+        image_key (str): Key for accessing images in the dataset.
+
+    Args:
+        name (Optional[str]): Name of the data loader instance.
+        dataset_config (str): Specifies the dataset config, defaulting to "train".
+        dataset_split (str): Specifies the dataset split, defaulting to "train".
+    """
+
+    label_key = "dx"
+    image_key = "image"
+    dataset_id = "marmal88/skin_cancer"
+    classification_label_mapping = {
+        "benign_keratosis-like_lesions": 0,
+        "basal_cell_carcinoma": 1,
+        "actinic_keratoses": 2,
+        "vascular_lesions": 3,
+        "melanocytic_Nevi": 4,
+        "melanoma": 5,
+        "dermatofibroma": 6,
+    }
+
+    def __init__(
+        self, name: Optional[str] = None, dataset_config: Optional[str] = None, dataset_split: str = "train"
+    ) -> None:
+        """
+        Initializes the `marmal88/skin_cancer` instance.
+
+        Args:
+            name (Optional[str]): Name of the data loader instance.
+            dataset_config (str): Specifies the dataset config, defaulting to "train".
+            dataset_split (str): Specifies the dataset split, defaulting to "train".
+        """
+        super().__init__(
+            hf_id=self.dataset_id,
+            hf_config=dataset_config,
+            hf_split=dataset_split,
+            name=name,
+        )
+
+        self.meta_exclude_keys.extend(
+            [
+                # Exclude input and output
+                self.image_key,
+                self.label_key,
+                # Exclude other info
+                "dx_type",
+                "image_id",
+                "lesion_id",
+            ]
+        )
```

Comment:
to be moved to `get_meta::flatten_dict::excludes`

Reply:
once this is moved, `__init__` is no longer needed to be overriden

Reply:
the other problem with get_meta from HF, is that the values are not correctly casted to the correct type, breaking the scan.

How did you resolve this issue for the HF bot? Do we need to provide a dict of types?